hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -1223,25 +1229,21 @@ def from_type(thing):
         return one_of(strategies)
     # If we don't have a strategy registered for this type or any subtype, we
     # may be able to fall back on type annotations.
-    # Types created via typing.NamedTuple use a custom attribute instead -
-    # but we can still use builds(), if we work out the right kwargs.
-    if issubclass(thing, tuple) and hasattr(thing, '_fields') \
-            and hasattr(thing, '_field_types'):
-        kwargs = {k: from_type(thing._field_types[k]) for k in thing._fields}
-        return builds(thing, **kwargs)
     if issubclass(thing, enum.Enum):
         assert len(thing), repr(thing) + ' has no members to sample'
         return sampled_from(thing)
-    # If the constructor has an annotation for all required arguments,
-    # or the class was defined with attrs, builds(thing) will work.
-    # (unless inference fails for an argument, but that's the same above too)
+    # If we know that builds(thing) will fail, give a better error message
     required = required_args(thing)
-    if not required or required.issubset(get_type_hints(thing.__init__)) or \
-            attr.has(thing):
-        return builds(thing)
-    # We have utterly failed, and might as well say so now.
-    raise ResolutionFailed('Could not resolve %r to a strategy; consider '
-                           'using register_type_strategy' % (thing,))
+    if not any([
+        not required,
+        required.issubset(get_type_hints(thing.__init__)),
+        attr.has(thing),
+        is_typed_named_tuple(thing)
```

Comment:
Why the additional check for `is_typed_named_tuple` here? You already got the argument names and type hints out of it, so I don't see what additional special cases are needed.